src/controllers/vstatefulset_controller/trusted/liveness_theorem.rs

```diff
@@ -0,0 +1,67 @@
+use crate::kubernetes_api_objects::spec::{prelude::*, pod_template_spec::*};
+use crate::kubernetes_cluster::spec::{cluster::*, message::*, controller::types::ReconcileLocalState};
+use crate::temporal_logic::defs::*;
+use crate::vstatefulset_controller::{
+    model::reconciler::*,
+    trusted::{spec_types::*, step::VStatefulSetReconcileStepView::*},
+    proof::predicate::*
+};
+use crate::vstd_ext::string_view::*;
+use vstd::prelude::*;
+
+verus !{
+
+pub open spec fn current_state_matches(vsts: VStatefulSetView) -> StatePred<ClusterState> {
+    |s: ClusterState| {
+        let owned_dyn_pods = s.resources().values().filter(owned_dyn_pods_filter(vsts));
+        let pvc_template_cnt = if vsts.spec.volume_claim_templates is Some {
+            vsts.spec.volume_claim_templates->0.len()
+        } else {0};
+        // 1. All needed pods exist and are up-to-date
+        &&& forall |ord: nat| #![trigger pod_name(vsts.metadata.name->0, ord)] 0 <= ord < vsts.spec.replicas.unwrap_or(1) ==> {
+            &&& exists |obj| {
```

Comment:
Do we need an existential quantifier here? This can be simpler by just saying that the pod name (constructed using the ordinal) exists in etcd, and the content of the pod matches the vsts. Similarly for pvc.